lib/client.go

```diff
@@ -27,28 +30,127 @@ type Client struct {
 	// ServerIp is the public IPv4 address of the server.
 	ServerIp netip.Addr
 
-	// Password is authenticates the client connection.
+	// Password authenticates the client connection.
 	Password string
 
 	// WgClient is a shared client for interacting with the WireGuard kernel module.
 	WgClient *wgctrl.Client
 
 	// Http is used to make connect requests to the server.
 	Http *http.Client
+
+	// serverPublicKey is the public key for the wireguard connection
+	serverPublicKey wgtypes.Key
+
+	// wgLink is the wireguard link
+	wgLink *linkWireguard
+
+	// wgCidr is the current wgCidr used with the wireguard link
+	wgCidr netip.Prefix
```

Comment:
- serverPublicKey does not need to be saved
- wgLink does not need to be saved
- wgCidr comment could clarify that it is the _current subnet assigned to the WireGuard interface_, if any.

Reply:
good points